### PR TITLE
tqdm logger: allow no metric_names

### LINKED ISSUE #285: tqm_logger: metric_names is currently not optional
Hi,

https://github.com/pytorch/ignite/blob/master/ignite/contrib/handlers/tqdm_logger.py#L75
This line should be modified to make `metric_names` optional. Here is a suggestion:
```
if metric_names is not None and not isinstance(metric_names, list):
    raise TypeError("metric_names should be a list, got {} instead".format(type(metric_names)))
```

Thanks

### PR DESCRIPTION
**Pull request by miguelvr**
Fixes #285

### CODE REVIEW

**Comment by vfdev-5** [comment]
@miguelvr can you also add a test for the argument ?

**Comment by miguelvr** [comment]
Sure thing

**Comment by miguelvr** [comment]
@vfdev-5 I'm trying to do something similar to what I've done with the other unit tests.

However, I'm getting weird behaviours.

The snippet for the code is like this:
```python
def test_pbar_no_metric_names(capsys):

    n_epochs = 2
    loader = [1, 2]
    engine = Engine(update_fn)

    pbar = ProgressBar()
    pbar.attach(engine)

    engine.run(loader, max_epochs=n_epochs)

    captured = capsys.readouterr()
    stderr = captured.err.split('\r')
    err = list(map(lambda x: x.strip(), stderr))
    err = list(filter(None, err))
    actual = err[-1]
    expected = u'Epoch 2: [1/2]  50%|█████      [00:00<00:00]'
    assert actual == expected
```

The comparison with the stderr string is behaving in a weird way and I'm obtaining a bunch of hexadecimal characters. However, if I debug and do everything manually the assertion checks out...

File: ignite/contrib/handlers/tqdm_logger.py [hunk 1]
```diff
@@ -47,6 +47,7 @@ def _update(self, engine, metric_names=None):
 
         if metric_names is not None:
             if not all(metric in engine.state.metrics for metric in metric_names):
+                self._close(engine)
```
Comment (miguelvr) [inline ignite/contrib/handlers/tqdm_logger.py]: this indeed seems to solve the problem, thanks!

**Comment by miguelvr** [comment]
LGTM!

**Pull request merged**